source/exe/hot_restart.h

```diff
@@ -138,8 +138,12 @@ class HotRestartImpl : public HotRestart,
   };
 
   struct RpcBase {
-    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase))
-        : type_(type), length_(length) {}
+    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase)) {
+      static_assert(std::is_standard_layout<RpcBase>::value, "is_standard_layout<RpcBase>");
```

Comment:
Yeah, good point re: the padding and different compilers. Sure, let's just clean it up and make it packed. We can just go back to your original fix and bump the hot restart/RPC version. Doing a full roll for us is fine, it just takes a long time. Let's do it.